prog/heatmap.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package prog
+
+import (
+	"bytes"
+	"fmt"
+)
+
+type Heatmap interface {
```

Comment:
It's hard to provide feedback on this w/o seeing how it will be used.
One question: will the mutation strategy also depend on the type of the data being mutated?
Another question: should ChooseLocation return a single int, or a range?
Methods like Populate are frequently done as "constructors" of the type, rather than interface methods. I.e. Heatmap has only ChooseLocation method, but then there is `func MakeGenericHeatmap(data []byte) Heatmap`.
The constructor allows to prevent uninitialized or double-initialized heatmaps. And also to pass implementation-specific args, e.g. `func MakeGenericHeatmap(data []byte, foo int) Heatmap`.

Reply:
Just to track this - recent versions of this PR address the issue regarding `Populate`, and choose the mutation strategy based on the type of data being mutated (though currently there is only one kind of compressed data, and one generic heatmap).